index.ts

```diff
@@ -14,18 +14,18 @@
  * to cleanly shut down before the process exits.
  */
 
-// TODO(rictic): convert this library to promises.
-
 /**
  * The type of a cleankill interrupt handler.
  */
-export type Handler = (doneHandling: () => void) => void;
+export type Handler = () => Promise<void>;
 
-let interruptHandlers: Handler[] = [];
+const interruptHandlers: Handler[] = [];
 
 /**
- * Register a handler to occur on SIGINT. All handlers are passed a callback,
- * and the process will be terminated once all handlers complete.
+ * Register a handler to occur on SIGINT. The handler must return a promise if
```

Comment:
Another way to structure this, which I'm kind of partial to, is to return a Promise from onInterrupt that resolves to an object that can cause exiting to wait. This way code that needs to synchronously clean up can just do:

```javascript
cleankill.onInterrupt().then((_) => {
  this.cleanup();
});
```

And code that needs to do async work can do:

```javascript
cleankill.onInterrupt().then((interrupt) => {
  interrupt.waitFor(this.cleanup());
});
```